test/e2e/specs/interactivity/directive-on-document.spec.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Internal dependencies
+ */
+import { test, expect } from './fixtures';
+
+test.describe( 'data-wp-on-document', () => {
+	test.beforeAll( async ( { interactivityUtils: utils } ) => {
+		await utils.activatePlugins();
+		await utils.addPostWithBlock( 'test/directive-on-document' );
+	} );
+
+	test.beforeEach( async ( { interactivityUtils: utils, page } ) => {
+		await page.goto( utils.getLink( 'test/directive-on-document' ) );
+	} );
+
+	test.afterAll( async ( { interactivityUtils: utils } ) => {
+		await utils.deactivatePlugins();
+		await utils.deleteAllPosts();
+	} );
+
+	test( 'callbacks should run whenever the specified event is dispatched', async ( {
+		page,
+	} ) => {
+		const counter = page.getByTestId( 'counter' );
+		await expect( counter ).toHaveText( '0' );
+		await page.keyboard.press( 'ArrowDown' );
+		await expect( counter ).toHaveText( '1' );
+	} );
+	test( 'the event listener is removed when the element is removed', async ( {
```

Comment:
@c4rl0sbr4v0, this test is flaky it failed 20+ times since PR got merged. Ref: #57983.

Reply:
PR that should fix it in https://github.com/WordPress/gutenberg/pull/58008